R/annotateProteinInfoFromIndra.R

```diff
@@ -0,0 +1,115 @@
+#' Annotate Protein Information from INDRA
+#'
+#' This function annotates a given dataframe with additional protein information 
+#' using various APIs. The input dataframe must contain columns 'Protein' and 'IDType'.
+#' The function adds new columns to the dataframe, including 'UniprotID', 'HgncID', 
+#' 'HgncName', 'IsKinase', 'IsPhosphatase', and 'IsTranscriptionFactor'.
+#'
+#' @param df A dataframe containing at least 'Protein' and 'IDType' columns.
+#' @return A dataframe with additional columns: 'UniprotID', 'HgncID', 'HgncName', 
+#' 'IsKinase', 'IsPhosphatase', and 'IsTranscriptionFactor'.
+#' @details 
+#' The function performs the following steps:
+#' \itemize{
+#'   \item Checks if the input dataframe contains 'Protein' and 'IDType' columns.
+#'   \item Initializes new columns ('UniprotID', 'HgncID', 'HgncName', 'IsKinase', 
+#'   'IsPhosphatase', 'IsTranscriptionFactor') as NA.
+#'   \item Maps 'Protein' to 'UniprotID' based on 'IDType'.
+#'   \item For 'Uniprot_Mnemonic' IDs, retrieves 'UniprotID' using an API call.
+#'   \item For valid 'UniprotID', retrieves 'HgncID' using an API call.
+#'   \item For valid 'HgncID', retrieves 'HgncName' using an API call.
+#'   \item For valid 'HgncName', retrieves characteristics ('IsKinase', 'IsPhosphatase', 
+#'   'IsTranscriptionFactor') using API calls.
+#' }
+#' @note The function relies on several internal API calls to retrieve the necessary 
+#' information. These API calls are represented by the following placeholder functions:
+#' \itemize{
+#'   \item \code{.callGetUniprotIdsFromUniprotMnemonicIdsApi}
+#'   \item \code{.callGetHgncIdsFromUniprotIdsApi}
+#'   \item \code{.callGetHgncNamesFromHgncIdsApi}
+#'   \item \code{.callIsKinaseApi}
+#'   \item \code{.callIsPhosphataseApi}
+#'   \item \code{.callIsTranscriptionFactorApi}
+#' }
```

Comment:
Have you gotten the chance to run `devtools::document()`?  I don't see the manual file associated with this documentation.

I'm also wondering if this note is necessary (assuming it's in the manual) since end users may not care about this.

Reply:
done